### ISSUE TITLE
Installation fails on clean machine

### ISSUE THREAD

**Issue by Volker-Weissmann** [pwithnall/dbus-deviation#18]
Hello, 

on a clean Arch-Linux machine:

```
# pip install dbus-deviation
Collecting dbus-deviation
  Downloading dbus-deviation-0.6.0.tar.gz (59 kB)
     |████████████████████████████████| 59 kB 1.7 MB/s 
    ERROR: Command errored out with exit status 1:
     command: /usr/bin/python -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-install-9ca6wvgl/dbus-deviation/setup.py'"'"'; __file__='"'"'/tmp/pip-install-9ca6wvgl/dbus-deviation/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' egg_info --egg-base /tmp/pip-pip-egg-info-s_3v_3hr
         cwd: /tmp/pip-install-9ca6wvgl/dbus-deviation/
    Complete output (45 lines):
    WARNING: The wheel package is not available.
    WARNING: The wheel package is not available.
      ERROR: Command errored out with exit status 1:
       command: /usr/bin/python -u -c 'import sys, setuptools, tokenize; sys.argv[0] = '"'"'/tmp/pip-wheel-4ygx8_7s/setuptools-pep8/setup.py'"'"'; __file__='"'"'/tmp/pip-wheel-4ygx8_7s/setuptools-pep8/setup.py'"'"';f=getattr(tokenize, '"'"'open'"'"', open)(__file__);code=f.read().replace('"'"'\r\n'"'"', '"'"'\n'"'"');f.close();exec(compile(code, __file__, '"'"'exec'"'"'))' bdist_wheel -d /tmp/pip-wheel-hfqrnlfm
           cwd: /tmp/pip-wheel-4ygx8_7s/setuptools-pep8/
      Complete output (6 lines):
      usage: setup.py [global_opts] cmd1 [cmd1_opts] [cmd2 [cmd2_opts] ...]
         or: setup.py --help [cmd1 cmd2 ...]
         or: setup.py --help-commands
         or: setup.py cmd --help
    
      error: invalid command 'bdist_wheel'
      ----------------------------------------
      ERROR: Failed building wheel for setuptools-pep8
    ERROR: Failed to build one or more wheels
    Traceback (most recent call last):
      File "/usr/lib/python3.9/site-packages/setuptools/installer.py", line 119, in fetch_build_egg
        subprocess.check_call(cmd)
      File "/usr/lib/python3.9/subprocess.py", line 373, in check_call
        raise CalledProcessError(retcode, cmd)
    subprocess.CalledProcessError: Command '['/usr/bin/python', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmplwtyu92w', '--quiet', 'setuptools_pep8']' returned non-zero exit status 1.
    
    The above exception was the direct cause of the following exception:
    
    Traceback (most recent call last):
      File "<string>", line 1, in <module>
      File "/tmp/pip-install-9ca6wvgl/dbus-deviation/setup.py", line 87, in <module>
        setup(
      File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 152, in setup
        _install_setup_requires(attrs)
      File "/usr/lib/python3.9/site-packages/setuptools/__init__.py", line 147, in _install_setup_requires
        dist.fetch_build_eggs(dist.setup_requires)
      File "/usr/lib/python3.9/site-packages/setuptools/dist.py", line 684, in fetch_build_eggs
        resolved_dists = pkg_resources.working_set.resolve(
      File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 766, in resolve
        dist = best[req.key] = env.best_match(
      File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1051, in best_match
        return self.obtain(req, installer)
      File "/usr/lib/python3.9/site-packages/pkg_resources/__init__.py", line 1063, in obtain
        return installer(requirement)
      File "/usr/lib/python3.9/site-packages/setuptools/dist.py", line 743, in fetch_build_egg
        return fetch_build_egg(self, req)
      File "/usr/lib/python3.9/site-packages/setuptools/installer.py", line 121, in fetch_build_egg
        raise DistutilsError(str(e)) from e
    distutils.errors.DistutilsError: Command '['/usr/bin/python', '-m', 'pip', '--disable-pip-version-check', 'wheel', '--no-deps', '-w', '/tmp/tmplwtyu92w', '--quiet', 'setuptools_pep8']' returned non-zero exit status 1.
    ----------------------------------------
ERROR: Command errored out with exit status 1: python setup.py egg_info Check the logs for full command output.
```

**Comment by pwithnall** [comment]
> ERROR: Failed to build one or more wheels

As per [this Stackoverflow answer](https://stackoverflow.com/a/44862371/2931197) you should be able to work around the problem by running `pip install wheel` first.

I’ll try and fix the problem properly in dbus-deviation soon.

**Comment by eli-schwartz** [comment]
I'm curious why pip is trying to install setuptools_pep8 at all; what's its purpose here? Why does this project require for the sake of building and installing it, that the user should add an optional interactive `python setup.py pep8` command as a linting tool that is not actually being used by the build process? Surely that should be a user choice, or added to a list of CI requirements rather than setup_requires?

In summary, why not revert commit a46110f36907a924d40a882003ba0bdcac09c71f?

**Comment by Volker-Weissmann** [comment]
> > ERROR: Failed to build one or more wheels
> 
> As per [this Stackoverflow answer](https://stackoverflow.com/a/44862371/2931197) you should be able to work around the problem by running `pip install wheel` first.
> 
> I’ll try and fix the problem properly in dbus-deviation soon.

Thank you, but I figured that out on my own.
